duckduckgo/Android
# Feature/craig/migrate sync to work manager

Task/Issue URL: https://app.asana.com/0/488551667048375/1147610969828115
Tech Design URL: 
CC: 

**Description**:
Migrates the app configuration sync to `WorkManager` and stops using `JobScheduler` APIs directly for it.

**Steps to test this PR**:
1. Checkout `develop`, and modify the times in `JobBuilder` to use ridiculously short times for `periodic` and `backoffCriteria` (like, a few milliseconds kind of short)
1. Install and launch. ensure the initial syncs finishes, so that a `Job` is scheduled with `JobScheduler`
1. Checkout this branch. Similar to before, set short values for periodic and backoff times in `AppConfigurationSyncWorkRequestBuilder`
1. Install and launch. Verify initial sync still happens when app first launches
1. Verify after app sync finishes, work is scheduled with `WorkManager`.
1. Wait for sync to happen again. Depending on the OS, this might take ~15mins ☕☕. Now is a good time to get Charles ready. We want to test failure path and success path, so you can set a breakpoint on one of the sync requests in Charles.
1. When the sync happens, kill a request in Charles to produce a failed sync. Verify it tries again, as per the retry policy
1. Let all requests succeed, and verify sync completes successfully 

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
